core/datainput.gms

```diff
@@ -239,6 +239,19 @@ if (c_ccsinjecratescen eq 3, sm_ccsinjecrate = sm_ccsinjecrate *   1.50 ); !! Up
 if (c_ccsinjecratescen eq 4, sm_ccsinjecrate = sm_ccsinjecrate * 200    ); !! remove flow constraint for DAC runs
 if (c_ccsinjecratescen eq 5, sm_ccsinjecrate = sm_ccsinjecrate *   0.20 ); !! sustainable estimate
 
+*** OR: regional reduction of ccs injection rate
+$ifThen.c_ccsinjecrateRegi not "%c_ccsinjecrateRegi%" == "off"
+Parameter
+  p_extRegiccsinjecrateRegi(ext_regi) "regional factor that limits CCS injection: 1 means no change, 0.6 means 40% reduction of CCS injection rate factor in respective region (extended regions)" / %c_ccsinjecrateRegi% /
```

Comment:
is there a reason, why you put the declaration after the if statement? I would prefer to always declare this parameter, also to avoid problems for fixing times and in the reporting

Reply:
good question. I did it as I did in the tax module, and I think that I discussed it with @Renato-Rodrigues then and there was a reason for it, but I do not remember exactly and maybe I misunderstood him. But I hope he can clarify.

Reply:
Placing the declaration inside the if statement is a way of not adding unnecessary content and size to the gdx file. 
It is usually more justifiable when you have a higher dimensional parameter, but I also don't see any issue on doing in this case. 
This would only throw an error in the fixings, if you are running an older remind version with a newer gdx. This is not good practice anyway, as you should instead in this case use the older gdx file (as the calibration for your model version was made for the old gdx), or update your code instead, in which case you wouldn't have any fixing error anyway.

Reply:
ON unrelated note, Why use a multiplicative factor in this case?
Wouldn't it be better to simply set the regions values directly. I think that using the multiplicative factor just creates confusion on understanding the actual values that is being applied to the regions that have the value overwritten.

Reply:
I would instead change the sm_ccsinjecrate to a parameter dependable of regi, assign the value of sm_ccsinjecrate to all regions by default, and then overwrite any region that containes values in the c_ccsinjecrateRegi.

This way we would have in the final gdx exactly the value that is being applied as injection rate directly in a parameter, instead of having to recalculate it from two different swicthes whenever we want to check results.

Reply:
so you mean. turn Scalar `sm_ccsinjecrate` into a parameter `sm_ccsinjecrate(regi)` and set it to the scenario value times the regional setting and that is it?

Reply:
to the first comment: wouldn't this throw an error also if you used a run that used the regional differentiation and therefore contains `p_extRegiccsinjecrateRegi` as fixing for a run that has it switched `off` and therefore never declared?

Reply:
yes, but I wouldn't set also sm_ccsinjecrate(regi) as a multiplicative term, but instead the absolute value. So:

pm_ccsinjecrate(regi) = sm_ccsinjecrate;
pm_ccsinjecrate(regi)$sm_ccsinjecrate(regi) = sm_ccsinjecrate(regi);

Reply:
also, if you want to be more general, you should use instead the ext_regi in the c_ccsinjecrateRegi switch. 
That way you can define values overwrites to region groups, instead of only remind regions, increasing compatibility with REMIND-21 regions. 
This allows you for example to set a value to "EUR_regi", instead of "EUR", and the same switch value would be compatible with REMIND-H12 and REMIND-21 region runs.

If you want a n example of how to use this let me know and I can show you quickly in a short call.

Reply:
> to the first comment: wouldn't this throw an error also if you used a run that used the regional differentiation and therefore contains `p_extRegiccsinjecrateRegi` as fixing for a run that has it switched `off` and therefore never declared?

Yes, but is this case even feasible or a design error? If your intention is to apply this as a baseline policy, this should never be disabled in the first place.
In this case simply remove any if conditional applied to it.
If you intend this as a policy scenario, then the error case you mention should not happen by logic design.

Reply:
1.
> If you intend this as a policy scenario, then the error case you mention should not happen by logic design.
Content-wise, I agree, but I prefer to try and write the code such that it doesn't break if somebody sets up scenarios that I haven't thought about

2.
> should use instead the ext_regi in the c_ccsinjecrateRegi switch.
That is what I do right now, don't I? Or do you mean I should maintain that even if I adapt the code to implement the other suggested changes?

3.
> I wouldn't set also c_ccsinjecrateRegi(regi) as a multiplicative term, but instead the absolute value
I'm not sure. I see the advantage of the advantage that you can set the injection rate explicitly. But: Let us say you generally want to have less CCS in Europe, but with different scenario settings. I think in this case, it is more transparent to have always `EUR_regi 0.9` in your scenario config instead of different values that than have to be compared to the scenario value in `default.cfg`. And if somebody ever changes the injection rates associated with `c_ccsinjecratescen`, then all the regi values have to be adapted to keep the order of the CCS injection rates fixed. But in the end, I think having it as absolute or multiplicative is rather a matter of taste. Maybe @LaviniaBaumstark can weigh in?

Reply:
2. Sorry about that, you are perfectly right. I didn't check the code in detail before making my comment.

3. While debugging, it is always preferable to have the final values used in the model directly available, then forcing the person debugging to check two parameters and to make a multiplication to figure out which value is being actually applied to the model as injection rate. That is why I don't think this is a matter of taste.
Also, I see as a undesirable effect changing a region specific injection rate if you change the global value.